src/main/java/uk/gov/hmcts/reform/prl/services/LocationRefDataService.java

```diff
@@ -119,6 +149,23 @@ private List<DynamicListElement> onlyEnglandAndWalesLocations(CourtDetails locat
             .map(this::getDisplayEntry).collect(Collectors.toList()));
     }
 
+    private List<DynamicListElement> filterOnboardedCourtList(String courtList, CourtDetails locationRefData) {
+        String[] filteredCourtArray = Arrays.stream(courtList.split(",")).filter(
+            element -> StringUtils.isEmpty(Arrays.stream(element.split(":")).toArray().length > 1
+                                               ? element.split(":")[1] : "")
+        ).toArray(size -> new String[size]);
+        return (locationRefData == null
+            ? new ArrayList<>()
+            : locationRefData.getCourtVenues().stream().filter(location -> !"Scotland".equals(location.getRegion()))
+            .filter(location -> FAMILY_COURT_TYPE_ID.equalsIgnoreCase(location.getCourtTypeId()))
+            .filter(location -> {
+                List<String> ids = Arrays.stream(filteredCourtArray).map(ele -> Arrays.stream(ele.split(":")).toArray()[0]
+                        .toString()).toList();
+                return ids.contains(location.getCourtEpimmsId());
+            })
```

Comment:
any specific reason for 3 filter conditions. can it be clubbed into one single condition

Reply:
@yogendraU Consolidated all three filter condition into one.